pkg/controllers/sidecar/config/config.go

```diff
@@ -98,16 +98,16 @@ func (s *ScyllaConfig) setupScyllaYAML() error {
 	cfg["rpc_address"] = "0.0.0.0"
 	cfg["endpoint_snitch"] = "GossipingPropertyFileSnitch"
 
-	overrideYAMLBytes, err := yaml.Marshal(cfg)
+	defaultYAMLBytes, err := yaml.Marshal(cfg)
 	if err != nil {
 		return errors.Wrap(err, "failed to parse override options for scylla.yaml")
 	}
-	scyllaYAMLConfigMapFilteredBytes, err := mergeYAMLs(scyllaYAMLConfigMapBytes, overrideYAMLBytes)
+	scyllaYAMLConfigMapFilteredBytes, err := mergeYAMLs(defaultYAMLBytes, scyllaYAMLConfigMapBytes)
 	if err != nil {
 		return errors.Wrap(err, "failed to merged config map YAML with default yaml values")
 	}
 
-	customScyllaYAMLBytes, err := mergeYAMLs(scyllaYAMLBytes, scyllaYAMLConfigMapFilteredBytes)
+	customScyllaYAMLBytes, err := mergeYAMLs(scyllaYAMLConfigMapFilteredBytes, scyllaYAMLBytes)
```

Comment:
This is wrong. This merges default scylla yaml into config map, resulting in overwiritng what user wants with default values. 
The correct merge should be: scylla.yaml <- operator presets <- user config map.

Reply:
As @zimnx stated the merging order is still not correct. This fixes it:

```
       defaultYAMLBytes, err := yaml.Marshal(cfg)
	if err != nil {
		return errors.Wrap(err, "failed to parse override options for scylla.yaml")
	}
	customScyllaYAMLBytes, err := mergeYAMLs(defaultYAMLBytes, scyllaYAMLBytes)

	if err != nil {
		return errors.Wrap(err, "failed to merged config map YAML with default yaml values")
	}

	scyllaYAMLConfigMapFilteredBytes, err := mergeYAMLs(customScyllaYAMLBytes, scyllaYAMLConfigMapBytes)
	if err != nil {
		return errors.Wrap(err, "failed to merged YAMLs")
	}

	// Write result to file
	if err = ioutil.WriteFile(scyllaYAMLPath, scyllaYAMLConfigMapFilteredBytes, os.ModePerm); err != nil {
		return errors.Wrap(err, "error trying to write scylla.yaml")
	}
```

Reply:
Partially right,
_defaultYAMLBytes_ is generated from the hardcoded config properties 
_scyllaYAMLConfigMapBytes_ is generated from _/mnt/scylla-config/scylla.yaml_  <-- user generated
and _scyllaYAMLBytes_ is generated from _/etc/scylla/scylla.yaml_
so line 105 is correct and only line 110 has to be switched around, correct?

Reply:
@nbenaglia Thanks, I followed the configuration naming wrong